src/core/packages/chrome/browser-internal/src/project_navigation/README.md

```diff
@@ -0,0 +1,112 @@
+|                            |    |
+|  +---------------------+   |    |     +-----------------+
+|  | activeNodes$        |---+----+---->| ChromeService   |
+|  +---------------------+   |    |     |                 |
+|                            |    |     |  +-----------+  |
++----------------------------+    +---->|  | navLinks$ |  |
+                                        |  +-----------+  |
+                                        |                 |
+                                        |  +-----------+  |
+                                        |  | collapsed$|  |
+                                        |  +-----------+  |
+                                        |                 |
+                                        +--------+--------+
+                                                 |
+                                                 | NavigationKibanaProvider
+                                                 | adapts streams
+                                                 v
+                                        +------------------+
+                                        | NavigationProvider|
+                                        |                  |
+                                        |  +-----------+   |
+                                        |  | Context   |   |
+                                        |  +-----------+   |
+                                        |                  |
+                                        +--------+--------+
+                                                 |
+                                                 | useNavigation() hook
+                                                 | consumes context
+                                                 v
+                                        +------------------+
+                                        | Navigation       |
+                                        | Component        |
+                                        |                  |
+                                        | - Renders UI     |
+                                        | - Highlights     |
+                                        |   active items   |
+                                        | - Shows          |
+                                        |   breadcrumbs    |
+                                        +------------------+
+```
+
+## Core Functions
+
+This service has three primary responsibilities:
+
+### 1. Consuming Navigation Components
+
+The service does not contain any UI. Instead, it processes navigation trees registered by plugins. This allows the navigation to be extended and customized by different parts of Kibana.
+
+-   **Plugins for Stateful Deployment**: Register navigation trees via the `addSolutionNavigation` service from `@kbn/shared-navigation-plugin`.
+-   **Serverless Plugins**: Register navigation trees via the `initNavigation` method from the serverless plugin's start contract.
```

Comment:
I think the reason for these two different methods / approaches was just because first we developed this for serverless, and then fitted / hacked for solution nav in stateless. I guess there is no reason to have these separate methods and we could unify later in the refactor